redis/client.py

```diff
@@ -2733,21 +2733,35 @@ def xpending_range(self, name, groupname, min, max, count,
         max: maximum stream ID.
         count: number of messages to return
         consumername: name of a consumer to filter by (optional).
+        idle: available from  version 6.2. filter entries by their
+        idle-time, given in milliseconds (optional).
         """
+        if {min, max, count} == {None}:
+            if idle is not None or consumername is not None:
+                raise DataError("if XPENDING is provided with idle time"
+                                " or consumername, it must be provided"
+                                " with min, max and count parameters")
+            return self.xpending(name, groupname)
+
         pieces = [name, groupname]
-        if min is not None or max is not None or count is not None:
-            if min is None or max is None or count is None:
-                raise DataError("XPENDING must be provided with min, max "
-                                "and count parameters, or none of them. ")
-            if not isinstance(count, int) or count < -1:
-                raise DataError("XPENDING count must be a integer >= -1")
-            pieces.extend((min, max, str(count)))
-        if consumername is not None:
-            if min is None or max is None or count is None:
-                raise DataError("if XPENDING is provided with consumername,"
-                                " it must be provided with min, max and"
-                                " count parameters")
-            pieces.append(consumername)
+        if min is None or max is None or count is None:
+            raise DataError("XPENDING must be provided with min, max "
+                            "and count parameters, or none of them.")
+        # idle
+        try:
+            if int(idle) < 0:
+                raise DataError("XPENDING idle must be a integer >= 0")
+            pieces.extend(['IDLE', idle])
+        except TypeError:
```

Comment:
What if the user uses a float number?@chayim 
It will pass the int test and redis server will raise an error: `redis.exceptions.ResponseError: value is not an integer or out of range`

Reply:
According to the redis docs IDLE should be an [int](https://redis.io/commands/XPENDING).  This means you have to catch ValueError as well to catch the int(str("5.5")) case.

```suggestion
        except (TypeError, ValueError):
```

Reply:
But I don't want to catch it...
I want to raise a Value error in case of int(5.5)

Reply:
Ok, I think I will leave it as before, and the command will throw a ResponseError